### PR TITLE
feat(kibbeh): add BaseOverlay component

### PR DESCRIPTION
**Pull request by lenilsondc**
This pull request adds the `BaseOverlay` component to the ui lib for the issue #637.

### CODE REVIEW

**Review by DGKSK8LIFE — APPROVED** [review]

File: kibbeh/src/ui/BaseOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      {actionButton && (
+        <div
+          className="w-full px-4 bg-primary-700 text-primary-100"
+          style={{
+            paddingTop: 8,
+            paddingBottom: 12,
+          }}
+        >
+          <button
+            className="cursos-pointer font-bold"
```
Comment (ajmnz) [inline kibbeh/src/ui/BaseOverlay.tsx]: Add `outline-none`

File: kibbeh/src/stories/BaseOverlay.story.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+);
+
+Notifications.bind({});
+
+export const Settings: Story<BaseOverlayProps> = ({
+  actionButton: actionLabel = "Log out",
+}) => (
+  <div style={{ width: 200 }}>
+    <BaseOverlay actionButton={actionLabel}>
+      <div className="py-2 flex flex-col">
```
Comment (ajmnz) [inline kibbeh/src/stories/BaseOverlay.story.tsx]: Remove `py-2`

**Review by ajmnz — CHANGES_REQUESTED** [review]

**Comment by lenilsondc** [comment]
Also, notice that the notification items has no natural size/padding, so I had to create a wrapper around it to look decent enough for the story.

File: kibbeh/src/stories/BaseOverlay.story.tsx (hunk 2 above)
Comment (lenilsondc) [reply kibbeh/src/stories/BaseOverlay.story.tsx]: Done

File: kibbeh/src/ui/BaseOverlay.tsx (hunk 1 above)
Comment (lenilsondc) [reply kibbeh/src/ui/BaseOverlay.tsx]: Done

**Comment by ajmnz** [comment]
> Also, notice that the notification items has no natural size/padding, so I had to create a wrapper around it to look decent enough for the story.

Went ahead and added some modifications.

**Review by ajmnz — APPROVED** [review]

**Pull request merged**